py/foundation/entity.py

```diff
@@ -486,3 +556,60 @@ def proto_to_entity(
   # This is a protocol decoding error, probably missing an if statement
   # in this function.
   raise AssertionError(f'Unhandled message type: {payload_type}')
+
+
+def entity_to_proto(entity: Entity) -> entity_pb2.Entity:
+  """Handles dispatching Entity serialization to subclasses of Entity.
+
+  In the proto definition, entity_pb2.Entity is a message containing a payload
+  which is any of the possible Entity types. This function wraps the result of
+  Entity.to_proto in the payload of entity_pb2.Entity.
+
+  Args:
+    entity: A Foundation Entity instance.
+  Returns:
+    A protobuf entity_pb2.Entity message.
+  Raises:
+    AssertionError: If not all payload types (defined in the .proto spec) are
+                    cased in this function.
+  """
+  payload = entity.to_proto()
+  if isinstance(payload, entity_pb2.OcrWord):
+    return entity_pb2.Entity(ocr_word=payload)
+  elif isinstance(payload, entity_pb2.Line):
+    return entity_pb2.Entity(line=payload)
+  elif isinstance(payload, entity_pb2.Paragraph):
+    return entity_pb2.Entity(paragraph=payload)
+  elif isinstance(payload, entity_pb2.TableCell):
+    return entity_pb2.Entity(table_cell=payload)
+  elif isinstance(payload, entity_pb2.TableRow):
+    return entity_pb2.Entity(table_row=payload)
+  elif isinstance(payload, entity_pb2.Table):
+    return entity_pb2.Entity(table=payload)
+  elif isinstance(payload, entity_pb2.Token):
+    return entity_pb2.Entity(token=payload)
+  elif isinstance(payload, entity_pb2.Phrase):
+    return entity_pb2.Entity(phrase=payload)
+  elif isinstance(payload, entity_pb2.Number):
+    return entity_pb2.Entity(number=payload)
+  elif isinstance(payload, entity_pb2.Integer):
+    return entity_pb2.Entity(integer=payload)
+  elif isinstance(payload, entity_pb2.Date):
+    return entity_pb2.Entity(date=payload)
+  elif isinstance(payload, entity_pb2.Time):
+    return entity_pb2.Entity(time=payload)
+  elif isinstance(payload, entity_pb2.Currency):
+    return entity_pb2.Entity(currency=payload)
+  elif isinstance(payload, entity_pb2.Name):
+    return entity_pb2.Entity(name=payload)
+  elif isinstance(payload, entity_pb2.Address):
+    return entity_pb2.Entity(address=payload)
+  elif isinstance(payload, entity_pb2.Cluster):
+    return entity_pb2.Entity(cluster=payload)
+  elif isinstance(payload, entity_pb2.GenericEntity):
+    return entity_pb2.Entity(custom=payload)
+
+  # This is a protocol decoding error, probably missing an if statement
+  # in this function.
+  raise AssertionError(f'Unhandled value: {type(payload).__name__}')
```

Comment:
Maybe a ValueError instead?

Reply:
I think this is more something that should really be checked statically, since mypy knows if all possible variants have been checked for or not. Like, if you forget to case a variant of PbEntityPayloadType, it's not a bad value it's a programmer error?

Reply:
I think the issue is that AssertionError should really only be raised when an actual assertion statement fails - if it's a programmer error, doesn't this mean they are passing an incorrect value into this function? I might just be nitpicking though!

Reply:
Yep, hopefully this PR https://github.com/instabase/foundation/pull/16 can make the intent of this check more clear.